### PR TITLE
Add g.version algorithm

### PR DESCRIPTION
**Pull request by AlisterH**
## Description

This allows the user to check the details of the GRASS version that the GRASS processing provider is using.  Useful for debugging.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Ideally all these values would become outputs from this algorithm (so eg a model can adapt dynamically). Would you be willing to implement that?

**Comment by AlisterH** [comment]
I'm not sure I understand what you're suggesting.
Whatever flags you use, g.version just produces one output.  So if you want separate outputs, one for copyright message, one for citation options, etc, you would run it multiple times with different flags, which sounds like a good job for a processing model.  But I'm not sure that would be useful for adaptable models.

Surely for adaptable models all you need is to:
- parse the grass version number (which you get by running this with all the options unchecked), and/or
- parse the output of  `g.extension -a`, which you get using g.extension.list algorithm that we committed not long ago.

My next thought was to look at a standard tool to check if a particular addon is available, and if not try to install it.  I think that would be most useful.

**Comment by AlisterH** [comment]
Ideally I guess what we should do for both g.version and g.extension.list is detect whether it is running as part of a model and if so store the results in an output variable rather than an output file.
At some point I might try to see if it is possible to do that using the ext file, but it may be over my head.

**Comment by AlisterH** [comment]
I did spend quite some time looking at _always_ saving the algorithm stdout to an output variable - I can just follow the existing code we're interacting with, but it's too complicated for me figure out the best way to do this.

**Comment by nyalldawson** [comment]
What I mean is that a html output is rather a dead end -- it's useful for a manual check of something, but restricts any use in automation (models or scripts). It would be better if we included the parsing in the algorithm so that the algorithm includes a string (or number) output value which is then super easy for use by these tools.

**Comment by AlisterH** [comment]
I've now committed changes to store the output in an output variable as well as to html.
I've also changed the way outputs are printed, as I don't think pformat was acceptable for these outputs.
This implementation may be contentious, but it works.

**Comment by AlisterH** [comment]
With this it is easy enough to get the actual version number:
```
processing.run("grass7:g.version",{'html': 'grass_version_info.html', '-r': 0, '-e':0})['MESSAGES'].split()[1]

'7.8.7'
```
If we can get #53246 merged, a QGIS plugin could provide an algorithm description for e.g. r.stream.order, and simply do this to ensure the GRASS addon is installed:
```
alglist=processing.run("grass7:g.extension.list",{'-a': 'TRUE', 'html': 'addons_list.html'})['MESSAGES']
if 'r.stream.order' not in alglist.splitlines():
 processing.run("grass7:g.extension.manage",{'extension': 'r.stream.order', 'operation': 0})
{}
```
Since we don't provide any output to show whether an addon install or uninstall was successful, it might be wise to then check again if it is actually installed...

**Comment by AlisterH** [comment]
Bumping so the bot doesn't kill us.

> I've now committed changes to store the output in an output variable as well as to html. I've also changed the way outputs are printed, as I don't think pformat was acceptable for these outputs. This aspect of the implementation may be contentious, but it works.

Also, I should note that there is something really inefficient about the code responsible for printing output messages to the algorithm dialog when we _aren't_ creating an html output.  Algorithms which produce a lot of output messages seem to speed up dramatically if generating an html output instead of printing to the algorithm dialog.

File: python/plugins/grassprovider/description/g.extension.list.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 g.extension
-g.extension.list - List GRASS addons.
+g.extension.list - List GRASS addons.<p>Select only one of the three options.
```
[the remark this answers is not in the document]
Comment (AlisterH) [reply python/plugins/grassprovider/description/g.extension.list.txt]: https://github.com/qgis/QGIS/pull/53635/commits/c18a5fec88f13b92d43e7488ab2dd4ed25086200

File: python/plugins/grassprovider/description/g.version.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+g.version
+g.version - Display GRASS GIS version info. <p>Prints only version if run with no options checked.
```
[the remark this answers is not in the document]
Comment (AlisterH) [reply python/plugins/grassprovider/description/g.version.txt:2]: I would need some more guidance on how to enable "multiple choices", but in any case, aren't checkboxes more user-friendly?

**Comment by AlisterH** [comment]
.

**Pull request closed without merging**